### PR TITLE
AA-402: send segment events for course-exit

### PR DESCRIPTION
**Pull request by mikix**
Specifically, a "I visited this page" event, with information on which variant was seen. And "I clicked this button" events, for our various calls to action.

https://openedx.atlassian.net/browse/AA-402

### CODE REVIEW

File: src/courseware/course/course-exit/CourseCelebration.jsx [hunk 1]
```diff
@@ -80,22 +80,16 @@ function CourseCelebration({ intl }) {
...
-      is_staff: administrator,
-      service,
-    });
-  };
-
   let buttonLocation;
   let buttonText;
   let buttonBackground = 'bg-white';
   let buttonVariant = 'outline-primary';
+  let buttonEvent = null;
```
Comment (Dillon-Dumesnil) [inline src/courseware/course/course-exit/CourseCelebration.jsx:88]: Why use `= null`? What happens if you use `let buttonEvent;` and it is never set? Undefined would be my guess. Is that okay?

File: src/courseware/course/course-exit/CourseCelebration.jsx [hunk 2]
```diff
@@ -80,22 +80,16 @@ function CourseCelebration({ intl }) {
...
   let certificateImage = certificate;
   let footnote;
   let message;
   let title;
+  let visitEvent = 'celebration_generic';
```
Comment (Dillon-Dumesnil) [inline src/courseware/course/course-exit/CourseCelebration.jsx:93]: Why set a value here if it is given a value in every possible case?

File: src/courseware/course/course-exit/utils.js [hunk 3]
```diff
@@ -68,4 +70,38 @@ function getCourseExitText(courseId, intl) {
   }
 }
 
-export { COURSE_EXIT_MODES, getCourseExitMode, getCourseExitText };
+// Meant to be used as part of a button's onClick handler.
+// For convenience, you can pass a falsy event and it will be ignored.
```
Comment (Dillon-Dumesnil) [inline src/courseware/course/course-exit/utils.js:74]: Curious, what is the use case for this?

File: src/courseware/course/course-exit/utils.js [hunk 4]
```diff
@@ -68,4 +70,38 @@ function getCourseExitText(courseId, intl) {
...
+  }
+
+  sendTrackEvent(`edx.ui.lms.course_exit.${event}.clicked`, {
+    course_id: courseId,
+    is_staff: administrator,
+  });
+};
+
+// Use like the following to call this only once on initial page load:
+// useEffect(() => logVisit(courseId, administrator, variant), [courseId, administrator, variant]);
```
Comment (Dillon-Dumesnil) [inline src/courseware/course/course-exit/utils.js:87]: :chefkiss:

**Review by Dillon-Dumesnil — APPROVED** [review]
LGTM. Couple small questions

File: src/courseware/course/course-exit/CourseCelebration.jsx (hunk 1 above)
Comment (mikix) [reply src/courseware/course/course-exit/CourseCelebration.jsx:88]: Yeah unset variables are undefined. I could have left it undefined. I guess I was just vibing a clearer "this page left blank intentionally" indicator.

File: src/courseware/course/course-exit/CourseCelebration.jsx (hunk 2 above)
Comment (mikix) [reply src/courseware/course/course-exit/CourseCelebration.jsx:93]: It's not given a value in every possible case. Unknown statuses (like if we get an null certData object) or audit-only end up as `generic`.

File: src/courseware/course/course-exit/utils.js (hunk 3 above)
Comment (mikix) [reply src/courseware/course/course-exit/utils.js:74]: If you have a complicated if/else/switch tree like we have and you might not set the event name for onClick. (In our case, we handle it everywhere. Or so I think. But defensively, we should check for an unset event name. And I think it makes more sense to do that inside the function than outside it.)

File: src/courseware/course/course-exit/CourseCelebration.jsx (hunk 2 above)
Comment (Dillon-Dumesnil) [reply src/courseware/course/course-exit/CourseCelebration.jsx:93]: But we gate on unknown statuses in CourseExit.jsx, right? Like you only enter CourseCelebration.jsx iff you have one of the statuses that match with the switch statement below.

File: src/courseware/course/course-exit/utils.js (hunk 3 above)
Comment (Dillon-Dumesnil) [reply src/courseware/course/course-exit/utils.js:74]: Yeah, I'm here for it. Was just curious.

File: src/courseware/course/course-exit/CourseCelebration.jsx (hunk 2 above)
Comment (mikix) [reply src/courseware/course/course-exit/CourseCelebration.jsx:93]: No, null certData goes to generic celebration (they are cert learner, but we don't know their status).

Plus, audit-only falls through and gets the generic variant.

Comment (Dillon-Dumesnil) [reply src/courseware/course/course-exit/CourseCelebration.jsx:93]: Oh dang. 👍

**Pull request merged**